### PR TITLE
[1LP][RFR] added new test_delete_vm_on_provider_side

### PR DESCRIPTION
**Pull request by lina-nikiforova**
## Purpose or Intent
Automated https://bugzilla.redhat.com/show_bug.cgi?id=1592430

### PRT Run
{{ pytest: -v cfme/tests/infrastructure/test_vm_delete.py::test_delete_vm_on_provider_side }}

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 25e097d4ab1add0a67dd1edbdee8fe55513ebf3a

The *local* fixture **`vm_test`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_delete.py**
    - *test_delete_vm_on_provider_side*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/infrastructure/test_vm_delete.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        assignee: anikifor
+        initialEstimate: 1/4h
+        casecomponent: Infra
+
+    Bugzilla:
+        1592430
+    """
+    logs = LogValidator("/var/www/miq/vmdb/log/evm.log", failure_patterns=[".*ERROR.*"])
+    logs.start_monitoring()
+    vm_test.mgmt.cleanup()
```
Comment (john-dupuy) [inline cfme/tests/infrastructure/test_vm_delete.py]: Q: Why not use `cleanup_on_provider()`? They're are functionally the same

File: cfme/tests/infrastructure/test_vm_delete.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    Polarion:
+        assignee: anikifor
+        initialEstimate: 1/4h
+        casecomponent: Infra
+
+    Bugzilla:
+        1592430
+    """
+    logs = LogValidator("/var/www/miq/vmdb/log/evm.log", failure_patterns=[".*ERROR.*"])
```
Comment (john-dupuy) [inline cfme/tests/infrastructure/test_vm_delete.py]: Q: Any specific error message we could search for here?

File: cfme/tests/infrastructure/test_vm_delete.py (hunk 1 above)
Comment (lina-nikiforova) [reply cfme/tests/infrastructure/test_vm_delete.py]: yes, I know that it's the same. My idea that it's more clear that delete happens on the provider's side. But looks like it was just me thinking that :) will change

File: cfme/tests/infrastructure/test_vm_delete.py (hunk 2 above)
Comment (lina-nikiforova) [reply cfme/tests/infrastructure/test_vm_delete.py]: No, in general, there should be no errors in the log for any reason. Even if there's some other reason for the error (not that vm was deleted), I'd much rather prefer the test to fail so I can investigate it.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 2255b830b5a4eab144b09172a8bd3daec52a5042

The *local* fixture **`vm_test`** is used in the following files:
- **cfme/tests/infrastructure/test_vm_delete.py**
    - *test_delete_vm_on_provider_side*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/infrastructure/test_vm_delete.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    logs.start_monitoring()
+    vm_test.cleanup_on_provider()
```
Comment (jawatts) [inline cfme/tests/infrastructure/test_vm_delete.py]: **Question** Do you think it is worth verifying that the vm was deleted?

Comment (lina-nikiforova) [reply cfme/tests/infrastructure/test_vm_delete.py]: @jawatts don't think so... 
`cleanup` method calls `delete` and it's the wait_for for that
```
    def delete(self):
        """
        Removes the entity on the provider
        """
        self.ensure_state(VmState.STOPPED)
        self.logger.debug(' Deleting RHEV VM %s/%s', self.name, self.uuid)

        self.api.remove()

        wait_for(
            lambda: not self.exists,
            message="wait for RHEV VM '{}' deleted".format(self.uuid),
            num_sec=300
        )
        return True
```

Comment (jawatts) [reply cfme/tests/infrastructure/test_vm_delete.py]: Yes! I thought so!

**Pull request merged**